Wikipedia/Code/WMFThemeableNavigationController.m

```diff
@@ -95,4 +100,31 @@ - (void)hideSplashViewAnimated:(BOOL)animated {
                      }];
 }
 
+- (void)_debounceTraitCollectionThemeUpdate {
+    if (@available(iOS 12.0, *)) {
+        WMFTheme *theme = self.traitCollection.userInterfaceStyle == UIUserInterfaceStyleLight ? WMFTheme.light : WMFTheme.black;
+        if (self.theme == theme) {
+            return;
+        }
+        for (UIViewController *vc in self.viewControllers) {
+            if (![vc isKindOfClass:[WMFAppViewController class]]) {
+                continue;
+            }
+            [(WMFAppViewController *)vc applyTheme:theme];
+        }
+    }
+}
+
+- (void)debounceTraitCollectionThemeUpdate {
+    if (@available(iOS 12.0, *)) {
+        [NSObject cancelPreviousPerformRequestsWithTarget:self selector:@selector(_debounceTraitCollectionThemeUpdate) object:nil];
+        [self performSelector:@selector(_debounceTraitCollectionThemeUpdate) withObject:nil afterDelay:0.3];
+    }
+}
+
+- (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection {
+    [super traitCollectionDidChange:previousTraitCollection];
+    [self debounceTraitCollectionThemeUpdate];
```

Comment:
Premature optimization but since WMFThemeableNavigationController is used throughout the app, maybe we should have a property flag for whether it responds to theme changes or not? Only 1 instance will have a WMFApplicationViewController right, seems like a waste for every WMFThemeableNavigationController to respond with this logic.

Reply:
Makes sense - maybe a delegate for receiving theme changes or trait collection changes that `WMFAppViewController` could set when it creates the root `WMFThemeableNavigationController` and it only responds here if that delegate property is set.

Reply:
Made this change - still WIP pending discussion of https://phabricator.wikimedia.org/T227123#5461131